AMI_Cleanup.py

```diff
@@ -0,0 +1,84 @@
+#Requires `pip install boto3`, `pip install python-dateutil`, and `pip install pytz`
+#Requires `aws configure` to use IAM credentials that can make the necessary AWS changes
+import boto3
+import shutil
+import datetime
+import dateutil.relativedelta
+import xml.etree.ElementTree as ET
+import subprocess
+import pytz
+
+#TODO: Jenkins is set up to use Justin IAM - change to beambox
+#This script needs to be run on the main jenkins server so that it can update the cloud AMI to use the latest
+#If there is ever a programmatic way to update the cloud config then this could be moved to a lambda
+#Until then it should be setup as a cron job per below - to be executed once every 2 days
+# sudo crontab -e
+# */* * 2 * * /home/ubuntu/AMI_Cleanup.py >> /home/ubuntu/AMI_Cleanup.log 2>&1
+
+utc=pytz.UTC
+
+def getEC2For(region):
+    ec2 = boto3.client('ec2', region_name=region)
+    return ec2
+
+def backup_and_retrieve_config():
+    today = datetime.datetime.today()
+    shutil.copy('/var/lib/jenkins/config.xml','/var/lib/jenkins/config.xml.bak_'+str(today.month)+'_'+str(today.day))
+    return ET.parse('/var/lib/jenkins/config.xml')
+
+def getTaggedAMIsAlongWithMostRecent(region):
+    ec2 = getEC2For(region)
+    amis = ec2.describe_images(Owners=['self'])
+    allAMIs = []
+    newestAMI = None
+    current_date_time = utc.localize(datetime.datetime.now())
+    for ami in amis['Images']:
+        if newestAMI is None:
+            newestAMI = ami
+        if 'beam-automation-' in ami['Name']:
+            ami_creation = dateutil.parser.parse(ami['CreationDate'])
+            current_newest_creation = dateutil.parser.parse(newestAMI['CreationDate'])
+            two_weeks_ago = current_date_time - dateutil.relativedelta.relativedelta(weeks=2)
+            if ami_creation < two_weeks_ago:
```

Comment:
Is keeping AMIs that are 2 weeks old fine? We update every couple of days currently.